### PR TITLE
Chrome package

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- setup new chrome client package
- add state manager (no need for redux or other libraries)
- add last visited pages client feature

### TODO
- [x] tests
- [ ] setup release

### CODE REVIEW

File: packages/chrome/src/ChromeProvider/chromeStateManager.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      ...attributes,
+    };
+
+    // update the subscribed clients
+    updateSubscriptions.forEach(({ onUpdate }) => {
+      onUpdate();
+    });
+  }
+
+  // lasy visited update event wrapper
```
Comment (BlakeHolifield) [inline packages/chrome/src/ChromeProvider/chromeStateManager.ts:59]:
```suggestion
  // last visited update event wrapper
```

**Review by BlakeHolifield — COMMENTED** [review]
Minor typo, but otherwise this looks sweet.

File: packages/chrome/src/ChromeProvider/ChromeProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import chromeState from './chromeState';
+import { get, post } from '../utils/fetch';
+
+const getLastVisited = () => get<string[]>('/api/chrome-service/v1/last-visited');
+
+const useLastPageVisitedUploader = (providerState: ReturnType<typeof chromeState>, chromeBackendEnabled?: boolean) => {
+  const { pathname } = useLocation();
+  useEffect(() => {
+    if (chromeBackendEnabled) {
+      post<string[], { pathname: string; title: string }>('/api/chrome-service/v1/last-visited', {
```
Comment (karelhala) [inline packages/chrome/src/ChromeProvider/ChromeProvider.tsx:15]: Can we make `'/api/chrome-service/v1/last-visited'` into a constant? Or even better use JS client?

File: packages/chrome/src/ChromeProvider/ChromeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }, [chromeBackendEnabled]);
+
+  return <ChromeContext.Provider value={providerState.current}>{children}</ChromeContext.Provider>;
+};
+
+export default ChromeProvider;
+
+<ChromeProvider>
+  <div>Foo bar</div>
+</ChromeProvider>;
```
Comment (karelhala) [inline packages/chrome/src/ChromeProvider/ChromeProvider.tsx:56]: Any reason for this random JSX snippet?

File: packages/chrome/src/utils/fetch.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+export function post<R, T extends Record<string, any> = Record<string, any>>(url: RequestInfo, data: T, options?: Omit<RequestInit, 'body'>) {
+  return fetch(url, {
+    ...options,
+    headers: {
+      ...options?.headers,
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
```
Comment (karelhala) [inline packages/chrome/src/utils/fetch.ts:14]: You should also add `credentials: 'include'` to include creds https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#sending_a_request_with_credentials_included

**Review by karelhala — COMMENTED** [review]
Looking good! Just a few cleaning comments, but overall looks good.

File: packages/chrome/src/ChromeProvider/ChromeProvider.tsx (hunk 2 above)
Comment (Hyperkid123) [reply packages/chrome/src/ChromeProvider/ChromeProvider.tsx:15]: The client is too heavy (you know no tree shaking). But yeah I will add a constant.

**Review by karelhala — APPROVED** [review]

**Pull request merged**